### PR TITLE
Improve debug sessions control

### PR DESCRIPTION
**Pull request by RedMickey**

### CODE REVIEW

File: src/extension/debugConfigurationProvider.ts [hunk 1]
```diff
@@ -7,6 +7,11 @@ import { Telemetry } from "../common/telemetry";
 import * as nls from "vscode-nls";
 const localize = nls.loadMessageBundle();
 
+export const DEBUG_TYPES = {
+    REACT_NATIVE: "reactnative",
```
Comment (SounD120) [inline src/extension/debugConfigurationProvider.ts:11]: I think we can use these debug types in the `DebugConfigProvider` below

File: src/extension/VSCodeSessionManager.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                connection.destroy();
+            }
+            this.connections.delete(terminateEvent.debugSession.id);
+        }
+
+        this.servers.delete(terminateEvent.debugSession.id);
+    }
+
+    public dispose() {
+        this.servers.forEach(server => server.close());
```
Comment (SounD120) [inline src/extension/VSCodeSessionManager.ts:53]: Should we close all connections the same way as servers?

File: src/extension/VSCodeSessionManager.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as vscode from "vscode";
+import * as Net from "net";
+import { DEBUG_TYPES } from "./debugConfigurationProvider";
+import { RNDebugSession } from "../debugger/rnDebugSession";
+import { DebugSessionBase, TerminateEventArgs } from "../debugger/debugSessionBase";
+import { DirectDebugSession } from "../debugger/direct/directDebugSession";
+
+export class VSCodeSessionManager implements vscode.DebugAdapterDescriptorFactory, vscode.Disposable {
```
Comment (SounD120) [inline src/extension/VSCodeSessionManager.ts:11]: Maybe it is better to name it ReactNativeSessionManager? What do you think?

Comment (RedMickey) [reply src/extension/VSCodeSessionManager.ts:11]: Yeah, sounds good, I'll rename this class.

**Pull request merged**